torch/utils/data/_utils/pin_memory.py

```diff
@@ -29,6 +29,8 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event, device):
     elif device == torch._C._get_privateuse1_backend_name():
         custom_device_mod = getattr(torch, torch._C._get_privateuse1_backend_name())
         custom_device_mod.set_device(device_id)
+    elif device is None:
+        torch.accelerator.set_device_idx(device_id)
```

Comment:
```suggestion
        torch.accelerator.set_device_index(device_id)
```
nit: we're using this new name to be more explicit